DFE-Digital/apply-for-qualified-teacher-status
# Add Maintenance Page Functionality

### Context

Show a friendly message to users when the service is down

### Changes proposed in this pull request

- following process [Maintenance page](https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/maintenance-page.md)
- Configure the maintenance page based on [the template](https://github.com/DFE-Digital/teacher-services-cloud/tree/main/templates/new_service/maintenance_page) for production and a test environment.
- Configure the [maintenance page workflow](https://github.com/DFE-Digital/teacher-services-cloud/blob/main/templates/new_service/.github/workflows/maintenance.yml).
- Test the maintenance page in a test environment.

### Guidance to review

i retested the workflow 
see https://github.com/DFE-Digital/apply-for-qualified-teacher-status/actions/runs/11609667671
and tested the maintenance page came up and app was available on temp url
and 
https://github.com/DFE-Digital/apply-for-qualified-teacher-status/actions/runs/11609705432
and tested the maintenance page now gone and app was available again

review changes and test doing the following: 
#### To test workflow - enable
```
get PIM Access to test

gh auth login

gh workflow run "Set maintenance mode" -r "origin/maintenance-page" -f environment=test -f mode=enable
```
wait a couple of mins 
```
https://apply-for-qts-temp.test.teacherservices.cloud - should show app

https://apply-for-qts-test-web.test.teacherservices.cloud/ - should show test page
```
#### To test workflow - disable
```
 gh workflow run "Set maintenance mode" -r "origin/maintenance-page" -f environment=test -f mode=disable
``
wait a couple of mins 
```
https://apply-for-qts-temp.test.teacherservices.cloud - should not exist

https://apply-for-qts-test-web.test.teacherservices.cloud/ - should show show app

```

#### To test makefile enable
```
get PIM Access to test

get a github token as per [Maintenance page](https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/maintenance-page.md)
```
make test enable-maintenance GITHUB_TOKEN=xxxxxxx
```
wait a couple of mins 
```
https://apply-for-qts-temp.test.teacherservices.cloud - should show app

https://apply-for-qts-test-web.test.teacherservices.cloud/ - should show test page

```
#### To test makefile disable
```
make test disable-maintenance GITHUB_TOKEN=xxxxxxx
```
wait a couple of mins 
```
https://apply-for-qts-temp.test.teacherservices.cloud - should not exist

https://apply-for-qts-test-web.test.teacherservices.cloud/ - should show show app

```

### Link to Trello card
[AFQTS: maintenance page](https://trello.com/c/2uMpMAnS/2100-afqts-maintenance-page)

### Checklist
